PapyrusPlugin/Sources/APIMacro.swift

```diff
@@ -70,24 +70,46 @@ extension FunctionDeclSyntax {
         let (method, path) = try apiMethodAndPath()
         try validateSignature()
 
-        let pathParameters = path.components(separatedBy: "/")
-            .compactMap { component in
-                if component.hasPrefix(":") {
-                    return String(component.dropFirst())
-                } else if component.hasPrefix("{") && component.hasSuffix("}") {
-                    return String(component.dropFirst().dropLast())
-                } else {
-                    return nil
-                }
+        func pathParameter(from component: String) -> String? {
+            if component.hasPrefix(":") {
+                return String(component.dropFirst())
+            } else if component.hasPrefix("{") && component.hasSuffix("}") {
+                return String(component.dropFirst().dropLast())
+            } else {
+                return nil
             }
+        }
+
+        let pathComponents = path.components(separatedBy: "/")
+        let pathParameters = pathComponents.compactMap(pathParameter)
 
         let attributes = parameters.compactMap({ $0.apiAttribute(httpMethod: method, pathParameters: pathParameters) })
         try validateAttributes(attributes)
 
+		let declaration = pathParameters.isEmpty ? "let" : "var"
         var buildRequest = """
-            var req = builder(method: "\(method)", path: "\(path)")
+            \(declaration) pathComponents: [String] = [\(
+                pathComponents
+                    .filter { !$0.isEmpty }
+                    .map { "\"\($0)\"" }
+                    .joined(separator: ", "))]
             """
 
+		for (index, component) in pathComponents.filter({ !$0.isEmpty }).enumerated() {
+            if let parameter = pathParameter(from: component) {
+                buildRequest += """
+                    if \(parameter) as Any? == nil {
+                        pathComponents.remove(at: \(index))
+                    }
+                    """
+            }
+        }
+
+        buildRequest += """
+           let path = pathComponents.joined(separator: "/")
+           var req = builder(method: \"\(method)\", path: path)
```

Comment:
Can we move this logic out of the `@API` macro and into the `RequestBuilder`?

I've been trying to keep all the "request construction" logic inside the `RequestBuilder` so that the generated code is dead simple to read.

Can make `RequestBuilder.parameters` a `[String: String?]` and then in the `parameterizedPath()` function exclude parameters that have `nil` values.